compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -247,5 +249,122 @@ template void hoistStaticallyBoundAllocationsInFunc<memref::AllocOp>(
 template void hoistStaticallyBoundAllocationsInFunc<memref::AllocaOp>(
     RewriterBase &rewriter, func::FuncOp funcOp);
 
+void analyzeSharedMemoryAlloc(func::FuncOp funcOp, ArrayRef<Operation *> allocs,
```

Comment:
Nit: rename to `analyseAllocsForCoalescing` or something more generic than a GPU specific name (i.e. `sharedMemory`)

Reply:
good catch, fixed it